SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs

```diff
@@ -0,0 +1,129 @@
+{
+    [Produces("application/json")]
+    [Route("api/Competition")]
+
+    public class CompetitionController : Controller
+    {
+        private ModelsContext _context;
+        public CompetitionController(ModelsContext db)
+        {
+            _context = db;
+        }
+ //----------------------------------------------------------------------------------------------------------------------------------------
+        /** The method to return an array of competition objects that hold the true Status*/
+        // GET: api/Competition
+        [HttpGet]
+        public IEnumerable<Competition> Get()
+        {
+            //get the competitions where(Status == true)
+            var competitionIDsQuery = from Comp in _context.Competitions
+                                      where Comp.Status == true
+                                      select Comp;
+            List<Competition> competitionsIDs = competitionIDsQuery.ToList<Competition>();
+            // return _context.Competitions.ToArray<Competition>();
+            return competitionsIDs;
+        }
+//----------------------------------------------------------------------------------------------------------------------------------------
+        //Getting the competition by ID
+        // GET: api/Competition/5
+        [HttpGet("{id}")]
+        public async Task<Competition> CompetitionGet(int id)
+        {
+            return await _context.Competitions.FindAsync(id);
+        }
+ //----------------------------------------------------------------------------------------------------------------------------------------
+        //Getting the competition by ID
+        // GET: api/Competition/5
+        [HttpGet("{Name}")]
+        public async Task<Competition> CompetitionGet(string Name)
+        {
+            return await _context.Competitions.FindAsync(Name);
+        }
+//----------------------------------------------------------------------------------------------------------------------------------------
+        //posting the competition to the competition table
+        // POST: api/Competition
+        [HttpPost]
+        public async Task<IActionResult> addCompetition([FromBody]Competition comp)
+        {
+            if (ModelState.IsValid)
+            {
+                Competition dbComp = null; //assume the competition does not exist
+
+                dbComp = _context.Competitions.FirstOrDefault(x => x.Name == comp.Name);
+                //if competition aready exist return
+                if (dbComp != null)
+                {
+                    return Ok("User already exists");
+                }
+                //Save the competition basic details
+                await _context.AddAsync(comp);
+                await _context.SaveChangesAsync();
+
+                return Ok("Competition saved successfully");
+            }
+            else
+            {
+                return new BadRequestObjectResult("competition cannot be null");
```

Comment:
This message might be confusing, it doesn't necessarily mean the competition is null here, it could be that any of the fields were invalid